features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/CheckableCompoundButtonMapper.kt

```diff
@@ -36,27 +40,95 @@ internal abstract class CheckableCompoundButtonMapper<T : CompoundButton>(
     @UiThread
     override fun resolveCheckableBounds(view: T, pixelsDensity: Float): GlobalBounds {
         val viewGlobalBounds = viewBoundsResolver.resolveViewGlobalBounds(view, pixelsDensity)
-        var checkBoxHeight = DEFAULT_CHECKABLE_HEIGHT_IN_PX
+        var checkBoxHeight = DEFAULT_CHECKABLE_HEIGHT_IN_DP * pixelsDensity.toLong()
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
             view.buttonDrawable?.let {
                 checkBoxHeight = it.intrinsicHeight.toLong()
             }
         }
-        // minus the padding
-        checkBoxHeight -= MIN_PADDING_IN_PX * 2
         checkBoxHeight = checkBoxHeight.densityNormalized(pixelsDensity)
         return GlobalBounds(
-            x = viewGlobalBounds.x + MIN_PADDING_IN_PX.densityNormalized(pixelsDensity),
+            x = viewGlobalBounds.x,
             y = viewGlobalBounds.y + (viewGlobalBounds.height - checkBoxHeight) / 2,
             width = checkBoxHeight,
             height = checkBoxHeight
         )
     }
 
+    override fun getCheckableDrawable(view: T): Drawable? {
+        val originCheckableDrawable = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            // drawable from [CompoundButton] can not be retrieved according to the state,
+            // so here two hardcoded indexes are used to retrieve "checked" and "not checked" drawables.
+            val checkableDrawableIndex = if (view.isChecked) {
+                CHECK_BOX_CHECKED_DRAWABLE_INDEX
+            } else {
+                CHECK_BOX_NOT_CHECKED_DRAWABLE_INDEX
+            }
+            (view.buttonDrawable?.constantState as? DrawableContainer.DrawableContainerState)?.getChild(
+                checkableDrawableIndex
+            )
+        } else {
+            // view.buttonDrawable is not available below API 23, so reflection is used to retrieve it.
```

Comment:
Actually a good idea, the thing is that, as i discussed with Nikita in the thread above, here the reflection logic is better to be kept as a class field for the performance issue, because calling a method of reflection for every wireframe creation can be time consuming. Based on that, I wonder if you still prefer to have a dedicated class with only static fields inside?